warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/handler/edge/ProtobufEdgeDataTypeHandler.java

```diff
@@ -805,134 +476,101 @@ private void setupPreconditionEvaluation(Multimap<String,NormalizedContentInterf
-    }
-
     protected void postProcessEdges(RawRecordContainer event, TaskInputOutputContext<KEYIN,? extends RawRecordContainer,KEYOUT,VALUEOUT> context,
-                    ContextWriter<KEYOUT,VALUEOUT> contextWriter, long edgesCreated, String loadDateStr) throws IOException, InterruptedException {}
+                    ContextWriter<KEYOUT,VALUEOUT> contextWriter, long edgesCreated, EdgeDataBundle edgeDataBundle) throws IOException, InterruptedException {}
 
-    protected EdgeDataBundle createEdge(EdgeDefinition edgeDef, RawRecordContainer event, NormalizedContentInterface ifaceSource, String sourceGroup,
-                    String sourceSubGroup, NormalizedContentInterface ifaceSink, String sinkGroup, String sinkSubGroup, String edgeAttribute2,
-                    String edgeAttribute3, Multimap<String,NormalizedContentInterface> normalizedFields,
-                    Map<String,Multimap<String,NormalizedContentInterface>> depthFirstList, String loadDate, long activityDate, boolean validActivityDate) {
-        // Get the edge value
-        EdgeDataBundle edgeDataBundle = new EdgeDataBundle(edgeDef, ifaceSource, ifaceSink, event, this.getHelper(event.getDataType()));
+    protected EdgeDataBundle createEdge(EdgeDataBundle edgeDataBundle, RawRecordContainer event, EdgeEventFieldUtil fieldUtil,
+                    NormalizedContentInterface ifaceSource, String sourceGroup, String sourceSubGroup, NormalizedContentInterface ifaceSink, String sinkGroup,
+                    String sinkSubGroup) {
 
-        if (edgeAttribute2 != null) {
-            edgeDataBundle.setEdgeAttribute2(edgeAttribute2);
-        }
-        if (edgeAttribute3 != null) {
-            edgeDataBundle.setEdgeAttribute3(edgeAttribute3);
-        }
-
-        edgeDataBundle.setLoadDate(loadDate);
-        edgeDataBundle.setActivityDate(activityDate);
-        edgeDataBundle.setValidActivityDate(validActivityDate);
-
-        // setup duration
-        if (edgeDef.hasDuration()) {
-            if (edgeDef.getUDDuration()) {
-                NormalizedContentInterface upnci = getNullKeyedNCI(edgeDef.getUpTime(), normalizedFields);
-                NormalizedContentInterface downnci = getNullKeyedNCI(edgeDef.getDownTime(), normalizedFields);
-                if (null != upnci && null != downnci) {
-                    edgeDataBundle.initDuration(upnci, downnci);
-                }
-            } else {
-                NormalizedContentInterface elnci = getNullKeyedNCI(edgeDef.getElapsedTime(), normalizedFields);
-                if (null != elnci) {
-                    edgeDataBundle.initDuration(elnci);
-                }
-            }
-        }
+        EdgeDefinition edgeDef = edgeDataBundle.getEdgeDefinition();
+        // Get the edge value
 
-        String typeName = getTypeName(event.getDataType());
+        VertexValue source = new VertexValue(edgeDef.isUseRealm(), edgeDef.getSourceIndexedFieldRealm(), edgeDef.getSourceEventFieldRealm(),
```

Comment:
For something like this consider just passing in the edge definition and allow the constructor to pull out what it needs.